e2e/cloudbuild.yaml

```diff
@@ -0,0 +1,200 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## This Cloud Build script is used to run the automated e2e test workflow
+
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+  id: Cache cloud-sdk image
+  waitFor: ['-']
+  args: ['pull', 'google/cloud-sdk:latest']
+  timeout: 120s
+- name: 'alpine'
+  id: Set up env.mk
+  waitFor: ['-']
+  entrypoint: /bin/ash
+  args:
+  - '-c'
+  - |
+      set -o errexit
+      set -o xtrace
+      set 
+      apk add gettext
+      cat e2e/env.mk.template | \
+      envsubst > env.mk
+  env:
+  - 'PROJECT_ID=${PROJECT_ID}'
+  - 'BUILD_ID=${_BUILD_ID_SHORT}'
+  timeout: 60s
+- name: 'google/cloud-sdk:latest'
+  id: Create DNS managed zone
+  waitFor: ['Cache cloud-sdk image']
+  args:
+  - gcloud
+  - dns
+  - managed-zones
+  - create
+  - ${_MANAGED_ZONE_NAME}
+  - --description="Managed zone for e2e testing commit ${SHORT_SHA}"
+  - --dns-name=${_DNS_NAME}
+  - --visibility=public
+  timeout: 60s
+- name: 'google/cloud-sdk:latest'
+  id: Export zone records
+  waitFor: ['Create DNS managed zone', 'Cache cloud-sdk image']
+  args:
+  - gcloud
+  - dns
+  - record-sets
+  - export
+  - --zone=${_MANAGED_ZONE_NAME}
+  - --zone-file-format
+  - ${_MANAGED_ZONE_NAME}.zonefile
+- name: 'alpine'
+  id: Remove SOA record
```

Comment:
I'm curious why this is necessary

Reply:
It's a bit of a confusing name -- this is removing the SOA record that got exported from the previous stage, because we don't want to import the SOA record into the top-level managed zone.

Reply:
Ahh interesting, that makes sense